src/components/Plans/PlanCalculator/BannerUpgrade/BannerUpgrade.js

```diff
@@ -0,0 +1,99 @@
+        link: `/plan-selection/premium/${URL_PLAN_TYPE[PLAN_TYPE.byEmail]}`,
+      };
+      const suggestionInfo = {
+        messageId: `plan_calculator.suggestion_for_${type.replace('-', '_')}`,
+        link: `/upgrade-suggestion-form`,
+      };
+
+      switch (type) {
+        case PLAN_TYPE.byCredit:
+        case PLAN_TYPE.byContact:
+          if (potencialUpgradePlans?.includes(PLAN_TYPE.byEmail)) {
+            return { banner: bannerInfo, suggestion: suggestionInfo };
+          } else {
+            const subscriberMonthlyPlan = {
+              ...suggestionInfo,
+              messageId: 'plan_calculator.advice_for_subscribers_large_plan',
+            };
+            return {
+              banner: subscriberMonthlyPlan,
+              suggestion: subscriberMonthlyPlan,
+            };
+          }
+
+        case PLAN_TYPE.byEmail:
+          return {
+            banner: {
+              ...bannerInfo,
+              link: `/upgrade-suggestion-form`,
+            },
+            suggestion: {
+              ...suggestionInfo,
+              link: `/plan-selection/premium/${URL_PLAN_TYPE[PLAN_TYPE.byEmail]}`,
+            },
+          };
+        default:
+          return {
+            banner: {
+              ...bannerInfo,
+              messageId: `plan_calculator.banner_for_unknown`,
+            },
+            suggestion: {
+              ...suggestionInfo,
+              messageId: `plan_calculator.suggestion_for_unknown`,
+            },
+          };
+      }
+    };
+    const sessionPlan = appSessionRef.current.userData.user;
+    const hightestPlan = currentPlanList.length === 1 && currentPlan.id === sessionPlan.idPlan;
+    const currentPlanSelected = currentPlan.id === currentPlanList[currentPlanList.length - 1].id;
+
+    if (currentPlanSelected && !hightestPlan) {
+      const bannerInfo = getUpgradeInfo(currentPlan.type);
+      return (
+        <div className="dp-calc-message" data-testid="dp-calc-message">
+          <p>
+            <FormattedMessage
+              id={bannerInfo.banner.messageId}
+              values={{
+                Link: (chunk) => <Link to={bannerInfo.banner.link}>{chunk}</Link>,
+              }}
+            />
+          </p>
+        </div>
+      );
+    } else if (currentPlanSelected && hightestPlan) {
+      const suggestionInfo = getUpgradeInfo(currentPlan.type);
+      return (
+        <div
+          className="dp-message-upgrade-plan m-t-24 m-b-24"
+          data-testid="dp-message-upgrade-plan"
+        >
+          <p>
+            <FormattedMessage
+              id={suggestionInfo.suggestion.messageId}
+              values={{
+                Link: (chunk) => <Link to={suggestionInfo.suggestion.link}>{chunk}</Link>,
+                Bold: (chunk) => <strong>{chunk}</strong>,
+              }}
+            />
+          </p>
+        </div>
+      );
+    } else {
+      return <></>;
+    }
```

Comment:
Veo que independientemente de si hightestPlan es true o es false, se ejecuta el método `getUpgradeInfo(currentPlan.type)`, crees que sería mas claro tratarlo de esta forma?

```
if (currentPlanSelected) {
      const upgradeInfo = getUpgradeInfo(currentPlan.type);

      if(!hightestPlan) {
        return (
          <div className="dp-calc-message" data-testid="dp-calc-message">
            <p>
              <FormattedMessage
                id={upgradeInfo.banner.messageId}
                values={{
                  Link: (chunk) => <Link to={upgradeInfo.banner.link}>{chunk}</Link>,
                }}
              />
            </p>
          </div>
        );
      }

      return (
        <div
          className="dp-message-upgrade-plan m-t-24 m-b-24"
          data-testid="dp-message-upgrade-plan"
        >
          <p>
            <FormattedMessage
              id={suggestionInfo.suggestion.messageId}
              values={{
                Link: (chunk) => <Link to={suggestionInfo.suggestion.link}>{chunk}</Link>,
                Bold: (chunk) => <strong>{chunk}</strong>,
              }}
            />
          </p>
        </div>
      );
    } else {
      return <></>;
      }
```

Reply:
Ok, it seems better to me 😉